### PR TITLE
Added dialog service, dialog component and DialogData interface

### LINKED ISSUE #103: Create shared service for open dialog

### PR DESCRIPTION
**Pull request by Gabriele-bil**
# Feature/Change Description

_Describe changes made here, link to any issue that is addressed_

# Developer Checklist

- [ ] Updated documentation or README.md
- [ ] If adding new feature(s), added and ran unit tests
- [ ] If create new release, bumped version number
- [x] Ran `npm run style:fix` for code style enforcement
- [x] Ran `npm run lint:fix` for linting
- [ ] Ran `npm audit` to discover vulnerabilities

### CODE REVIEW

File: src/app/shared/services/open-dialog/open-dialog.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+describe('OpenDialogService', () => {
+  let service: OpenDialogService
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({})
+    service = TestBed.inject(OpenDialogService)
+  })
+
+  xit('should be created', () => {
```
Comment (Gabriele-bil) [inline src/app/shared/services/open-dialog/open-dialog.service.spec.ts]: Sembrerebbe che i test falliscano a causa delle dependecy injection di material

**Review by CosNig — APPROVED** [review]

**Review by CarmeloB4 — APPROVED** [review]

**Review by FanniFelfoldi — APPROVED** [review]

**Review by Mattewn99 — APPROVED** [review]

**Review by Ste91wq — APPROVED** [review]

File: src/app/app.module.ts [hunk 2]
```diff
@@ -7,10 +7,11 @@ import { AppRoutingModule } from './app-routing.module'
 import { AppComponent } from './app.component'
 import { AdvertisementModule } from './modules/core/advertisement/advertisement.module'
 import { MaterialModule } from './modules/shared/material.module'
+import { DialogComponent } from './shared/components/dialog/dialog.component'
 import { PageNotFoundComponent } from './shared/components/pagenotfound/pagenotfound.component'
 
 @NgModule({
-  declarations: [AppComponent, PageNotFoundComponent],
+  declarations: [AppComponent, PageNotFoundComponent, DialogComponent],
```
Comment (pc-codingavision) [inline src/app/app.module.ts]: The `DialogComponent` imported in the app module looks quite a bad idea to me. It should go in a shared module most probably.

File: src/app/shared/components/dialog/dialog.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    }).compileComponents()
+  }))
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(DialogComponent)
+    component = fixture.componentInstance
+    fixture.detectChanges()
+  })
+
+  xit('should create', () => {
```
Comment (pc-codingavision) [inline src/app/shared/components/dialog/dialog.component.spec.ts:23]: Excluding test cases is definitely NOT a way to implement test. Please fix it.

File: src/app/shared/models/dialog-data.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+export interface IDialogData {
+  h1: string
```
Comment (pc-codingavision) [inline src/app/shared/models/dialog-data.ts]: `h1`.....really?!? 😒

File: src/app/shared/models/dialog-data.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+  h2: string
```
Comment (pc-codingavision) [inline src/app/shared/models/dialog-data.ts]: `h2`.....really?!? 😒

File: src/app/shared/models/dialog-data.ts [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
+  p: string
```
Comment (pc-codingavision) [inline src/app/shared/models/dialog-data.ts]: `p`.....really?!? 😒

File: src/app/shared/services/open-dialog/open-dialog.service.spec.ts (hunk 1 above)
Comment (pc-codingavision) [reply src/app/shared/services/open-dialog/open-dialog.service.spec.ts]: Andrebbe fixato in qualche maniera ;)

File: src/app/shared/services/open-dialog/open-dialog.service.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import { Injectable } from '@angular/core'
+import { MatDialog, MatDialogRef } from '@angular/material/dialog'
+
+import { IDialogData } from './../../models/dialog-data'
+import { DialogComponent } from '../../components/dialog/dialog.component'
+
+@Injectable({
+  providedIn: 'root',
+})
+export class OpenDialogService {
```
Comment (pc-codingavision) [inline src/app/shared/services/open-dialog/open-dialog.service.ts]: `OpenDialogService`, not really the best name possible....is it?

**Review by pc-codingavision — CHANGES_REQUESTED** [review]

**Review by pc-codingavision — APPROVED** [review]

**Pull request merged**